### DISCUSSION TITLE
Noob questions

### DISCUSSION THREAD

**Discussion by Kuroonehalf** [focus-editor/focus#222] [Q&A]
Hi. I've been tinkering around with the program and I dig how clean it looks and how easy it is to tweak the config. However, I am having some trouble doing basic things. I think this program borrows a lot from VIM or something, right? And I've never really used that, or get how people navigate it. So I'm having trouble doing stuff like:

- How do I quickly get to my project's given directory without manually browsing to it? I have it open on my file manager, I can copy the full path, but pasting it on the Open File dialog doesn't seem to work. Am I missing something obvious?
- How can I reopen the last open file/project?
- How do I navigate between files? I'm used to programs like notepad++ that have tabs. How would I switch back and forth between two files here? Do I have to go Open File, and navigate back and forth through the dialog every time?

I think those are the big ones. Apologies if these are kind of dumb, but I am struggling. Would appreciate clarification. Thanks.

**Comment by focus-editor** [comment]
Hi. The things you're talking about are not 100% done and will be improved over time.

The main expected workflow right now is this:

First, you add directories to the workspace in the config. 
Then, the following commands can be used to open the file dialog in different modes:
```
Ctrl-P                      show_open_file_dialog_in_search_mode
Ctrl-O                      show_open_file_dialog_in_navigate_mode
Ctrl-Shift-O                show_open_file_dialog_in_navigate_mode_from_root
```
You either look up your file name in the search mode dialog (similar to VSCode/Sublime), or navigate to files using the navigate mode dialog, which should already have your project directories in there.

> How can I reopen the last open file/project?

After editor restart you can't currently do that, except switching to your project and then following the normal workflow.

> How do I navigate between files?

Through the means described above. Tabs are probably not going to be added to the base editor. You can go back to previous editor state using these commands which should help with navigation.

```
Alt-Minus                   move_to_previous_editor_history
Alt-Plus                    move_to_next_editor_history

Ctrl-PageUp                 move_to_previous_buffer
Ctrl-PageDown               move_to_next_buffer
```

That said, the workflow will be improved over time. But probably still not by adding tabs.

**Reply by Kuroonehalf** [reply]
I see. Thank you for the clarification. A few things I'd like to ask / follow up on:

- Like in the example above, it'd be nice if navigate mode was able to interpret a full path pasted from clipboard. I sometimes want to work on a project temporarily and it'd be nice if I could just grab the path, launch focus, ctrl-o, paste, and be on my way. Manually browsing from root is a pain, especially if it's a folder somewhere deep in the desktop on windows or something. And having to pull out the config file to add the path also feels unnecessarily involved for a short time use thing.
- After playing with search mode and navigate mode a bit more, I could see myself getting used to this as a fine replacement to tabs. 
- However, swapping back and forth between two files through the buffer shortcuts still feels a bit wonky. When I move to the previous buffer and then forward again, it seems to lose the scroll position I had, making it cumbersome for comparing code. If it could remember that it'd be great. Remembering scroll position per file is a feature I'd really like to have here. In my game projects I have my dialog script files mixed with my code files, and a feature like this makes it great to pick up where I left off with minimal friction.

**Reply by focus-editor** [reply]
Thank you for your feedback. 
I agree that pasting a path would be a good feature - please create a separate feature request so that it doesn't get lost.
Going back and forth in editor history will be improved, currently it's just cursor history going in large steps, that's why scrolling is lost.
